protocol/xmrmaker/net.go

```diff
@@ -99,22 +103,30 @@ func (inst *Instance) HandleInitiateMessage(msg *net.SendKeysMessage) (net.SwapS
 		return nil, nil, errOfferIDNotSet
 	}
 
+	if err := coins.ValidatePositive("providedAmount", msg.ProvidedAmount); err != nil {
+		return nil, nil, err
+	}
+
 	offer, offerExtra, err := inst.offerManager.GetOffer(msg.OfferID)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	providedAmount := offer.ExchangeRate.ToXMR(msg.ProvidedAmount)
+	providedAmount, err := offer.ExchangeRate.ToXMR(msg.ProvidedAmount)
+	if err != nil {
+		return nil, nil, err
+	}
 
-	if providedAmount < offer.MinAmount {
+	if providedAmount.Cmp(offer.MinAmount) < 0 {
+		// TODO: This message will be confusing to the end-user, since they provided ETH, not XMR
```

Comment:
I think it's fine since it's returned to the taker (ETH-provider) saying the amount they provided is too low?

Reply:
The error messages are fixed and I removed the TODO.